server/controllers/homepageController.ts

```diff
@@ -52,4 +71,391 @@ export default class HomepageController {
       return res.redirect(`${config.serviceUrls.digitalPrisons}/global-search/results?searchText=${name}`)
     }
   }
+
+  getServices(req: CustomRequest, res: Response, next: NextFunction) {
+    const { whereaboutsMaintenanceMode } = config.app
+    const { keyworkerMaintenanceMode } = config.app
+
+    let whereaboutsConfig
+    if (whereaboutsMaintenanceMode) {
+      whereaboutsConfig = { enabled: false }
+    } else {
+      whereaboutsConfig = { enabled: true }
+    }
+
+    let keyworkerPrisonStatus
+    if (keyworkerMaintenanceMode) {
+      keyworkerPrisonStatus = { migrated: false } // this can be empty because we're using the feature flag in getTasks
+    } else {
+      keyworkerPrisonStatus = { migrated: true }
+    }
+
+    const allServicess = this.getTasks(
+      res.locals.user.activeCaseLoadId,
+      res.locals.user.locations,
+      res.locals.user.staffId,
+      whereaboutsConfig,
+      keyworkerPrisonStatus,
+      res.locals.user.userRoles,
+    )
+
+    return allServicess
+  }
+
+  getTasks(
+    activeCaseLoadId: string,
+    locations: Location[],
+    staffId: string,
+    whereaboutsConfig: WhereAboutsConfig,
+    keyworkerPrisonStatus: KeyWorkerPrisonStatus,
+    userRoles: string[],
+  ) {
+    const userHasRoles = (roles: string[]) => hasAnyRole(userRoles, roles)
```

Comment:
See previous comments.
Seems confusing to create your own function with the same name as the existing function from `utils.ts`
I think it's better to call your new function something else.